cdc/sink/cdclog/file.go

```diff
@@ -197,7 +197,7 @@ func (f *fileSink) flushLogMeta() error {
 		return cerror.WrapError(cerror.ErrMarshalFailed, err)
 	}
 	// FIXME: if initialize succeed, O_WRONLY is enough, but sometimes it will failed
-	file, err := os.OpenFile(f.logPath.meta, os.O_CREATE|os.O_WRONLY, defaultFileMode)
+	file, err := os.OpenFile(f.logPath.meta, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, defaultFileMode)
```

Comment:
`O_TRUNC` may discard all contents of a file, if TiCDC is killed between line 200 and line 201, the log.meta is lost.
I suggest using a temp file eg "log.meta.tmp" for saving the data, after that, we can rename the temp file to log.meta.

Note: rename is atomic, https://man7.org/linux/man-pages/man2/rename.2.html#:~:text=atomically

Reply:
Thanks for your advice. I'll do it.